sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2132,7 +2134,12 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         val kvs = units.indices.map { i =>
           val u = units(i).getText
           val v = if (values(i).STRING() != null) {
-            string(values(i).STRING())
+            val value = string(values(i).STRING()).trim
+            if (!intervalValStrPattern.matcher(value).matches()) {
```

Comment:
how about `!value.forall { c => c == '.' || Character.isDigit(c)}`? We don't have to use regex.

Reply:
ok maybe the first version is better, as we basically just want to avoid the `value` string to contain unit. `value.exists(Character.isLetter)` should be good enough.

Reply:
And we should add comments to explain why we need the check.